prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py

```diff
@@ -44,10 +45,21 @@ def check_static_variable_run_status(
             return output
 
     @staticmethod
-    def replace_generic_string_value(prompt: str, variable: str, value: str) -> str:
-        replaced_prompt = prompt.replace(variable, value)
+    def replace_generic_string_value(prompt: str, variable: str, value: Any) -> str:
+        formatted_value: str = value
+        if not isinstance(value, str):
+            formatted_value = VariableService.handle_json_and_str_types(value)
```

Comment:
NIT: We are checking if its not an instance of str type but we end up calling a function named `VariableService.handle_json_and_str_types()` which doesn't help improve the understanding without stepping into the code. Try to improve the naming maybe

Reply:
@chandrasekharan-zipstack Noted the suggestion. Will pick it up in later optimizations.